src/math/math_base.h

```diff
@@ -105,5 +106,64 @@ struct PointBase3D : public PointBase2D<_Type>
     _Type z;
 };
 
+template <typename _Type>
+struct LineBase2D
+{
+    LineBase2D( const PointBase2D<_Type> & _p1, const PointBase2D<_Type> & _p2 )
+        : p1( _p1 ), p2( _p2 )
+    { }
+
+    LineBase2D( const PointBase2D<_Type> & _p, double angleRadians )
```

Comment:
I thought about it but we then also allow integral types as `angleRadians`. Do we want it?

Reply:
We couldn't cover situations but we have a good solution for this as concrete implementation of this function for specific type.

Logically only float and double types are used to represent a line while integral (like int, char and etc) are not really useful. At least I can't find any case for this...

Reply:
To explain further: not always necessary to use `double` just because it gives higher accuracy, sometimes `float` is enough. At least setting a type `float` for the angle in `float` implementation is reasonable.

Reply:
So we use _Type?

Reply:
Yes, I'm suggesting this.

Reply:
ok :)